### ISSUE TITLE
[DefaultCodegen] Incorrect name for request body parameter of array types 

### ISSUE THREAD

**Issue by etherealjoy** [OpenAPITools/openapi-generator#376]
##### Description

When the logs below show up 
```
[main] ERROR codegen.DefaultCodegen - String to be sanitized is null. Default to ERROR_UNKNOWN
[main] ERROR codegen.DefaultCodegen - String to be sanitized is null. Default to ERROR_UNKNOWN
```
There is an issue in sanitizing the name of the parameter. 
For example for aspnetcore the output is below
```C#
[SwaggerOperation("CallEndpoint")]
public virtual IActionResult CallEndpoint([FromBody]List<string> ERROR_UNKNOWN)
```

One of the issues with this is related to how arrays in a request body are handled in `DefaultCodegen`
```java
public CodegenParameter fromRequestBody(RequestBody body, Map<String, Schema> schemas, 
 Set<String> imports, String bodyParameterName)
```

the function is called from 
```java
public CodegenOperation fromOperation(String path,
				  String httpMethod,
				  Operation operation,
				  Map<String, Schema> schemas,
				  OpenAPI openAPI)
```
where
```java
String bodyParameterName = "";
if(op.vendorExtensions != null 
&& op.vendorExtensions.containsKey("x-codegen-request-body-name")) 
{
	bodyParameterName = (String) op.vendorExtensions.get("x-codegen-request-body-name");
}
bodyParam = fromRequestBody(requestBody, schemas, imports, bodyParameterName);
```
and then in `fromRequestBody`
```java
...
 else if (ModelUtils.isArraySchema(schema)) 
...
while (innerCp != null) {
	if (innerCp.complexType != null) {
		imports.add(innerCp.complexType);
	}
	mostInnerItem = innerCp;
	innerCp = innerCp.items;
}

if (StringUtils.isEmpty(bodyParameterName)) {
	codegenParameter.baseName = mostInnerItem.complexType;
} else {
	codegenParameter.baseName = bodyParameterName;
}
codegenParameter.paramName = toArrayModelParamName(codegenParameter.baseName);
```
`mostInnerItem.complexType` is null for arrays of primitive or maps therefore here in `toArrayModelParamName` sanitization happens with a `null` resulting in NPE for generators which don't overload `toVarName` properly.

**TL;DR;**

Problem is here 
`codegenParameter.baseName = mostInnerItem.complexType;`
 below is incorrect
```java
            while (innerCp != null) {
                if (innerCp.complexType != null) {
                    imports.add(innerCp.complexType);
                }
                mostInnerItem = innerCp;
                innerCp = innerCp.items;
            }

            if (StringUtils.isEmpty(bodyParameterName)) {
                codegenParameter.baseName = mostInnerItem.complexType;
            } else {
                codegenParameter.baseName = bodyParameterName;
            }
```

##### openapi-generator version

`master`
##### OpenAPI declaration file content or url

```yaml
---
# This is a sample Swagger spec
 
swagger: "2.0"
info:
  description: Container Application Model Description
  version: 1.0.0
  title: ContainerApplication
  
host: localhost:32000
basePath: /api/v1.0
schemes:
- http

tags:  
- name: ApplicationEndpointTest
  x-displayName: "ApplicationEndpointTest"
  description: |
    "Test of endpoints "
    
paths:
  /default/EndpointTest: 
    post:
      tags:
      - ApplicationEndpointTest      
      description: "post a request"
      operationId: callEndpoint
      consumes:
      - application/json
      parameters:
      - name: "requestList"
        in: "body"
        description: "Requests to post"
        required: true
        schema:
          $ref: "#/definitions/listitems"
      responses:
        200:
          description: "Successful response"
        404:
          description: "The request was not successfully executed."
      
  
definitions:
  listitems : 
    description: "list of items"
    type: array
    items: 
      type: "string"
```
##### Command line used for generation

```shell
java -jar ~/openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar \
generate -i ./api.yaml -g aspnetcore -o genserver 
```
##### Related issues/PRs

#373 
##### Suggest a fix/enhancement

This part is handled correctly in case of array of Models but for the primitive and arrays it causes a warning log which can trigger NPE in some generators and causes a wrong parameter name to be created.

**Comment by jmini** [comment]
Internally we are using Swagger-Parser to read the model. If we read an OAS2, the tool also do an on-the-fly conversion “OAS2 -> OAS3”.

Here is what we get from the parser when we read your example spec:

```yaml
openapi: 3.0.1
info:
  title: ContainerApplication
  description: Container Application Model Description
  version: 1.0.0
servers:
- url: http://localhost:32000/api/v1.0
tags:
- name: ApplicationEndpointTest
  description: |
    "Test of endpoints "
  x-displayName: ApplicationEndpointTest
paths:
  /default/EndpointTest:
    post:
      tags:
      - ApplicationEndpointTest
      description: post a request
      operationId: callEndpoint
      requestBody:
        description: Requests to post
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/listitems'
        required: true
      responses:
        200:
          description: Successful response
          content: {}
        404:
          description: The request was not successfully executed.
          content: {}
components:
  schemas:
    listitems:
      type: array
      description: list of items
      items:
        type: string
```

In OAS3 the Request body does not have a name...

I have raised https://github.com/swagger-api/swagger-parser/issues/748 in order to keep this information during the conversion. Feel free to indicate there that you are interested by the feature.

**Comment by jmini** [comment]
PR #1721 was merged.

@etherealjoy can you verify if your issue is fixed with the latest `4.0.0-SNAPSHOT`?

**Comment by etherealjoy** [comment]
@jmini 
I will check and update.
